File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int curr = 0;
        int maxi = INT_MIN;
        
        for(int num: nums) {
            
            curr = max(curr + num, num); 
            maxi = max(maxi, curr);
        }
    
        return maxi; 
    }
};

File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        
        int sum = 0;
        int maxi = Integer.MIN_VALUE;
        
        for(int x : nums )
        {
            sum += x;
            maxi = Math.max(sum,maxi);
            if(sum < 0)
                sum = 0;
        }
        
        return maxi;
    }
}